source/partials/drupal-8-convert-to-composer.md

```diff
@@ -0,0 +1,255 @@
+
+- The script does not resolve patches.
+
+  Many times a module will be "patched" or have a `.patch` file that fixes known issues before the fix is available in the downloaded version. This script does not attempt to resolve any patches.
+
+If you still want to try it:
+
+1. Say the following out loud: "Pantheon is not responsible for what I am about to do."
+
+1. Create a directory called `bin` within your repository and copy the `migrateModules.php` file from [GitHub](https://github.com/pantheon-upstreams/drupal-project/pull/8/files#diff-c68470275758ca395b98d53ed258b63519435d492dd51531c4cf372814c6593e) into that directory.
+
+1. To run the module migration script, `cd` to the root directory of the repository and run `bin/migrateModules.php`.
+
+1. To see which modules were added, run `git diff composer.json`.
+
+1. Remove the `composer.lock` file. Composer will create a new one with the modules you just added.
+
+1. Run `composer install` and resolve any remaining version conflicts.
+
+</Accordion>
+-->
+
+#### Libraries
+
+Libraries can be handled similarly to modules, but the specifics depend on how your library code was included in the source site. If you're using a library's API, you may have to do additional work to ensure that library functions properly.
+
+### Custom Code
+
+Manually copy custom code from the existing site repository to the Composer-managed directory.
+
+#### Modules and Themes
+
+Modules:
+
+```bash{promptUser:user}
+mkdir -p ../$SITE-composer/web/modules/custom # create the directory if it doesn't already exist
+cp -r modules/custom/awesome_module ../$SITE-composer/web/modules/custom
+```
+
+Themes:
+
+```bash{promptUser:user}
+mkdir -p ../$SITE-composer/web/themes/custom # create the directory if it doesn't already exist
+cp -r themes/custom/great_theme ../$SITE-composer/web/themes/custom
+```
+
+Follow suit with any other custom code you need to carry over.
+
+#### Settings.php
+
+Your existing site may have customizations to `settings.php` or other configuration files. Review these carefully and extract relevant changes from these files to copy over. Always review any file paths referenced in the code, as these paths may change in the transition to Composer.
+
+We don't recommend that you completely overwrite the `settings.php` file with the old one, as it contains customizations for moving the configuration directory you don't want to overwrite, as well as platform-specific customizations.
+
+```bash{promptUser:user}
+git status # Ensure working tree is clean
+git show master:sites/default/settings.php > web/sites/default/original-settings.php
+diff -Nup --ignore-all-space web/sites/default/settings.php web/sites/default/original-settings.php
+# edit web/sites/default/settings.php and commit as needed
+rm web/sites/default/original-settings.php
+```
```

Comment:
I'm not sure whether we should do anything about the config directory, e.g. it could maybe be set like this:
```
$settings['config_sync_directory'] = dirname(dirname(dirname(__DIR__))) . '/config';
```

Reply:
This would have to go in settings.php after settings.pantheon.php is included. Maybe we should fix the Pantheon settings file instead.

Reply:
@greg-1-anderson can you clarify what we should do here?